sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.submitApplication.e2e-spec.ts

```diff
@@ -0,0 +1,364 @@
+  ApplicationStatus,
+  EducationProgramOffering,
+  OfferingIntensity,
+  ProgramYear,
+  Student,
+} from "@sims/sims-db";
+import { addDays, getISODateOnlyString } from "@sims/utilities";
+import { SaveApplicationAPIInDTO } from "../models/application.dto";
+import { SystemUsersService } from "@sims/services";
+import { FormNames, FormService } from "../../../services";
+import { AppStudentsModule } from "../../../app.students.module";
+import { createFakeSFASPartTimeApplication } from "@sims/test-utils/factories/sfas-part-time-application";
+
+describe("ApplicationStudentsController(e2e)-submitApplication", () => {
+  let app: INestApplication;
+  let appModule: TestingModule;
+  let appDataSource: DataSource;
+  let sharedStudent: Student;
+  let db: E2EDataSources;
+  let programYear: ProgramYear;
+  let systemUsersService: SystemUsersService;
+  let formService: FormService;
+  const PROGRAM_YEAR_PREFIX = 1990;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    appDataSource = dataSource;
+    db = createE2EDataSources(dataSource);
+    sharedStudent = await saveFakeStudent(db.dataSource);
+    systemUsersService = nestApplication.get(SystemUsersService);
+    // Program Year for the following tests.
+    programYear = await ensureProgramYearExists(db, PROGRAM_YEAR_PREFIX);
+    formService = await getProviderInstanceForModule(
+      appModule,
+      AppStudentsModule,
+      FormService,
+    );
+    // To check the study dates overlap error, the BYPASS_APPLICATION_SUBMIT_VALIDATIONS needs to be set false.
+    process.env.BYPASS_APPLICATION_SUBMIT_VALIDATIONS = "false";
+  });
+
+  it("Should throw study dates overlap error when an application submitted for a student via the SIMS system has overlapping study start or study end dates with another application.", async () => {
+    // Arrange
+    // First offering created with start date 10 days prior to the
+    // current date and end date 10 days ahead of the current date.
+    const firstApplicationOfferingInitialValues = {
+      studyStartDate: getISODateOnlyString(addDays(-10)),
+      studyEndDate: getISODateOnlyString(addDays(10)),
+      offeringIntensity: OfferingIntensity.partTime,
+    } as EducationProgramOffering;
+    await saveFakeApplication(
+      appDataSource,
+      {
+        student: sharedStudent,
+        programYear,
+      },
+      { offeringInitialValues: firstApplicationOfferingInitialValues },
+    );
+    // Second offering created with start date as the current
+    // date and the end date 20 days ahead of the current date.
+    // This will result in the offering start date of this offering
+    // overlap with the first offering's start and end dates.
+    const secondApplicationOfferingInitialValues = {
+      studyStartDate: getISODateOnlyString(new Date()),
+      studyEndDate: getISODateOnlyString(addDays(20)),
```

Comment:
Please make sure that data prepared in a test is not in relevance to to other tests. Using a separate student in test will help achieving it.